contracts/src/v0.8/ccip/pools/GHO/UpgradeableLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,277 @@
+    _transferOwnership(owner);
+
+    s_router = IRouter(router);
+
+    // Pool can be set as permissioned or permissionless at deployment time only to save hot-path gas.
+    if (i_allowlistEnabled) {
+      _applyAllowListUpdates(new address[](0), allowlist);
+    }
+    s_bridgeLimit = bridgeLimit;
+  }
+
+  /// @notice Locks the token in the pool
+  /// @param amount Amount to lock
+  /// @dev The whenHealthy check is important to ensure that even if a ramp is compromised
+  /// we're able to stop token movement via ARM.
+  function lockOrBurn(
+    address originalSender,
+    bytes calldata,
+    uint256 amount,
+    uint64 remoteChainSelector,
+    bytes calldata
+  )
+    external
+    virtual
+    override
+    onlyOnRamp(remoteChainSelector)
+    checkAllowList(originalSender)
+    whenHealthy
+    returns (bytes memory)
+  {
+    // Increase bridged amount because tokens are leaving the source chain
+    if ((s_currentBridged += amount) > s_bridgeLimit) revert BridgeLimitExceeded(s_bridgeLimit);
+
+    _consumeOutboundRateLimit(remoteChainSelector, amount);
+    emit Locked(msg.sender, amount);
+    return "";
+  }
+
+  /// @notice Release tokens from the pool to the recipient
+  /// @param receiver Recipient address
+  /// @param amount Amount to release
+  /// @dev The whenHealthy check is important to ensure that even if a ramp is compromised
+  /// we're able to stop token movement via ARM.
+  function releaseOrMint(
+    bytes memory,
+    address receiver,
+    uint256 amount,
+    uint64 remoteChainSelector,
+    bytes memory
+  ) external virtual override onlyOffRamp(remoteChainSelector) whenHealthy {
+    // This should never occur. Amount should never exceed the current bridged amount
+    if (amount > s_currentBridged) revert NotEnoughBridgedAmount();
+    // Reduce bridged amount because tokens are back to source chain
+    s_currentBridged -= amount;
+
+    _consumeInboundRateLimit(remoteChainSelector, amount);
+    getToken().safeTransfer(receiver, amount);
+    emit Released(msg.sender, receiver, amount);
+  }
+
+  /// @notice returns the lock release interface flag used for EIP165 identification.
+  function getLockReleaseInterfaceId() public pure returns (bytes4) {
+    return LOCK_RELEASE_INTERFACE_ID;
+  }
+
+  // @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) public pure virtual override returns (bool) {
+    return
+      interfaceId == LOCK_RELEASE_INTERFACE_ID ||
+      interfaceId == type(ILiquidityContainer).interfaceId ||
+      super.supportsInterface(interfaceId);
+  }
+
+  /// @notice Gets Rebalancer, can be address(0) if none is configured.
+  /// @return The current liquidity manager.
+  function getRebalancer() external view returns (address) {
+    return s_rebalancer;
+  }
+
+  /// @notice Sets the Rebalancer address.
+  /// @dev Only callable by the owner.
+  function setRebalancer(address rebalancer) external onlyOwner {
+    s_rebalancer = rebalancer;
```

Comment:
Any reason to not have an event here? (same applicable for `setRateLimitAdmin()`

Reply:
just because the standard CCIP implementation does not have it and want to minimize changes as much as possible